server/src/main/java/org/elasticsearch/search/retriever/RetrieverBuilder.java

```diff
@@ -181,15 +184,27 @@ protected static RetrieverBuilder parseInnerRetrieverBuilder(XContentParser pars
 
     protected String retrieverName;
 
+    /**
+     * Determines if this retriever contains sub-retrievers that need to be executed prior to search.
+     */
+    public boolean isCompound() {
+        return false;
+    }
```

Comment:
was it required that this method gets introduced in this PR, or could it be added later as a followup?

Reply:
I'd like to make this PR about introducing a compound mode for retriever in the search infra. The followup is to add a compound retriever, which is another big pr in itself.

Reply:
The pr is smaller now and is just about adding the compound mode.